storage/mysql/log_storage_test.go

```diff
@@ -980,31 +981,39 @@ func TestGetUnsequencedCounts(t *testing.T) {
 	for i := int64(1); i < 10; i++ {
 		// Put some leaves in the queue of each of the logs
 		for j, logID := range logIDs {
-			tx := beginLogTx(s, logID, t)
-			defer tx.Close()
-			numToAdd := i + int64(j)
-			leaves := createTestLeaves(numToAdd, expectedCount[logID])
-			if _, err := tx.QueueLeaves(ctx, leaves, fakeDequeueCutoffTime); err != nil {
-				t.Fatalf("Failed to queue leaves: %v", err)
-			}
-			commit(tx, t)
-			expectedCount[logID] += numToAdd
+			j, logID := j, logID
+			func() {
+				tx := beginLogTx(s, logID, t)
+				defer tx.Close()
```

Comment:
One suggestion might be something like a `func runWithTX(s, logID, f func(tx) err) err` and you'd do here:
```go
for ... {
  err := runWithTX(s, logID, func f(tx ...) err {
      numToAdd ..
      ...
      if bad {
         return errors.New(...) 
      }
      ...
      return nil
    })
    if err != nil {
      t.Error...
    }
}
```

and `runWithTX` does the
```go
tx := beginLog...
defer tx.Close()
return f(tx)
```

Reply:
Yes, the funcs were because of the defer-close. I went with your first suggestion and explicitly closed the tx.

I have been meaning to add `runInTX` methods for a while now, but in a slightly different form. I'll save those to another PR.